packages/svelte/src/compiler/phases/2-analyze/visitors/RenderTag.js

```diff
@@ -49,8 +53,37 @@ export function RenderTag(node, context) {
 	) {
 		e.render_tag_invalid_call_expression(node);
 	}
+	
+	const parent = context.path.at(-2);
+	const is_animated = snippet?.body.nodes.some(is_animate_directive);
+
+	if(is_animated) {
+		if (parent?.type !== 'EachBlock') {
+			e.animation_invalid_placement(node);
+		}
+		else if (!parent.key) {
+			e.animation_missing_key(parent);
+		} else if (parent.body.nodes.filter((n) =>
```

Comment:
Same logic as in [phases/2-analyze/visitors/shared/element.js#L92](https://github.com/sveltejs/svelte/blob/main/packages/svelte/src/compiler/phases/2-analyze/visitors/shared/element.js#L92). Maybe re-use logic for consistent maintainence for future updates to this logic?